src/util/utils.js

```diff
@@ -1,8 +1,74 @@
 /* eslint-disable max-classes-per-file */
+const http = require('http');
+const https = require('https');
+const { Resolver } = require('dns').promises;
+const fetch = require('node-fetch');
+
 const util = require('util');
 const logger = require('../logger');
 const stats = require('./stats');
 
+const resolver = new Resolver();
+// Cloudflare and Google dns
+resolver.setServers(['1.1.1.1', '8.8.8.8']);
+
+const LOCALHOST_IP = '127.0.0.1';
+const LOCALHOST_URL = `http://localhost`;
+
+const staticLookup = (versionid) => async (hostname, _, cb) => {
+  let ips;
+  const resolveStartTime = new Date();
+  try {
+    ips = await resolver.resolve(hostname);
+  } catch (error) {
+    stats.timing('fetch_dns_resolve_time', resolveStartTime, { versionid, error: true });
+    logger.info(`unable to resolve IP address for ${hostname}`);
+    cb(null, `unable to resolve IP address for ${hostname}`, 4);
+    return;
+  }
+  stats.timing('fetch_dns_resolve_time', resolveStartTime, { versionid });
+  logger.info(`resolved ${hostname} to ${ips}`);
+
+  if (ips.length === 0) {
+    cb(null, `resolved empty list of IP address for ${hostname}`, 4);
+    return;
+  }
+
+  for (const ip of ips) {
+    if (ip.includes(LOCALHOST_IP)) {
+      cb(null, `cannot use ${LOCALHOST_IP} as IP address`, 4);
+      return;
+    }
+  }
+
+  cb(null, ips[0], 4);
+};
+
+const staticDnsAgent = (scheme, versionId) => {
+  const httpModule = scheme === 'http' ? http : https;
```

Comment:
If I pass any other scheme like tcp, ftp etc, it will be interpreted as https by the system ?

Reply:
only http or https allowed